src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,41 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectTodo: Todo;
+  setSelectTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({ selectTodo, setSelectTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [hasLoading, setHasLoading] = useState(false);
```

Comment:
The state variable `hasLoading` should be renamed to `isLoading` to better reflect its purpose as a boolean flag indicating the loading state.